acquia_lift.install

```diff
@@ -639,6 +638,186 @@ function acquia_lift_update_7017() {
+  // First organize our results into arrays of agents with their option sets.
+  $agents = array();
+  foreach ($result as $row) {
+    $options = unserialize($row->options);
+    $agents[$row->machine_name][] = array(
+      'agent_name' => $row->machine_name,
+      'agent_label' => $row->label,
+      'osid' => $row->osid,
+      'decision_name' => empty($row->decision_name) ? 'osid-' + $row->osid : $row->decision_name,
+      'option_count' => count($options),
+      'options' => $options,
+      'targeting' => unserialize($row->targeting),
+    );
+  }
+  // If an agent has more than one option set or if its option set contains fixed
+  // targeting rules, it cannot be converted and will be treated as a legacy agent.
+  $legacy_agents = $convertibles = array();
+  // If personalize module is not enabled all campaigns will be treated as legacy.
+  $personalize_enabled = module_exists('personalize');
+  foreach ($agents as $agent_name => $details) {
+    $option_set = reset($details);
+    if (!empty($option_set['targeting']) || !$personalize_enabled) {
+      $legacy_agents[] = $agent_name;
+      continue;
+    }
+    if (count($details) > 1) {
+      // We can convert this if all option sets have the same decision name.
+      $decision_name = '';
+      $option_count = 0;
+      $supported = TRUE;
+      foreach ($details as $i => $option_set) {
+        if ($i == 0) {
+          $decision_name = $option_set['decision_name'];
+          $option_count = $option_set['option_count'];
+        }
+        else {
+          // If any of the option sets has a different decision name or option
+          // count, then this agent cannot be converted.
+          if ($option_set['decision_name'] != $decision_name || $option_set['option_count'] != $option_count) {
+            $supported = FALSE;
+          }
+        }
+      }
+      if ($supported) {
+        $convertibles[$agent_name] = $details;
+      }
+      else {
+        $legacy_agents[] = $agent_name;
+      }
+    }
+    else {
+      $convertibles[$agent_name] = $details;
+    }
+  }
+  // If personalize module is not enabled we just want to save the legacy agents
+  // and be done.
+  if (!$personalize_enabled && !empty($legacy_agents)) {
+    drupal_set_message(t('Personalize module was disabled so existing campaigns could not be converted to the new style personalizations. They will be treated as legacy campaigns.'), 'warning');
+    variable_set('acquia_lift_legacy_agents', $legacy_agents);
+    return;
+  }
+
+  // Now convert anything that can be converted.
+  foreach ($convertibles as $original_agent_name => $details) {
+    // Create a new acquia_lift_target agent which will delegate to this one.
+    $agent = new stdClass();
+    $agent->label = 'Parent personalization for ' . $details[0]['agent_label'];
```

Comment:
This ends up really long and repetitive in the unibar.... maybe just "Converted: " or something simpler?